packages/duf/src/evo/data_converters/duf/common/duf_wrapper.py

```diff
@@ -0,0 +1,243 @@
+#  Copyright © 2025 Bentley Systems, Incorporated
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import os
+from collections import defaultdict
+
+import evo.data_converters.duf.common.deswik_types as dw
+import evo.logging
+
+logger = evo.logging.getLogger("data_converters")
+
+
+class InvalidDUFFileException(ValueError):
+    def __init__(self, message):
+        super().__init__(message)
+
+
+class DUFFileNotFoundException(FileNotFoundError):
+    def __init__(self, message):
+        super().__init__(message)
+
+
+class ObjectCollector:
+    def __init__(self, verbose=False):
+        self._objs: dict[dw.Category, dict[type, dw.BaseEntity]] = defaultdict(lambda: defaultdict(list))
+        self._layers_by_guid: dict[dw.Guid, dw.Layer] = {}
+        self._verbose = verbose
+
+    def Loaded(self, category, item):
+        if category == dw.Category.Layers:
+            if not isinstance(item, dw.Layer):
+                raise TypeError(f"Expected Layer, got {type(item).__name__}")
+            self._layers_by_guid[item.Guid] = item
+        elif hasattr(item, "Layer"):
+            # Sometimes the loaded layer on the object is not the same as the one in the layer collection, fix this
+            if item.Layer is not None and item.Layer is not self._layers_by_guid[item.Layer.Guid]:

```

Comment:
If a layer were not found for some reason should it raise a particular kind of exception to fail gracefully or is the default TB ok?